derex/runner/settings/derex/base.py

```diff
@@ -1,190 +1,44 @@
-# It will be overridden later, if an `lms_site_name` variable has been specified in the config
-SITE_NAME = {"lms": "localhost:4700", "cms": "localhost:4700"}[SERVICE_VARIANT]
-
-STATIC_ROOT_BASE = "/openedx/staticfiles"
-COMPREHENSIVE_THEME_DIRS.append("/openedx/themes")  # type: ignore
-STATIC_ROOT_BASE = os.environ.get("STATIC_ROOT_LMS", "/openedx/staticfiles")
-STATIC_ROOT = {  # type: ignore
-    "lms": path(STATIC_ROOT_BASE),
-    "cms": path(STATIC_ROOT_BASE) / "studio",
-}[os.environ.get("SERVICE_VARIANT")]
-WEBPACK_LOADER["DEFAULT"]["STATS_FILE"] = (  # type: ignore
-    STATIC_ROOT / "webpack-stats.json"
-)
-
-
-MEDIA_ROOT = "/openedx/media"
-VIDEO_TRANSCRIPTS_SETTINGS["location"] = MEDIA_ROOT  # type: ignore  # noqa
-VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"] = MEDIA_ROOT  # type: ignore  # noqa
-PROFILE_IMAGE_BACKEND["options"]["location"] = MEDIA_ROOT  # type: ignore  # noqa
-COMPREHENSIVE_THEME_DIRS.append(Path("/openedx/themes"))  # type: ignore  # noqa
-# STATICFILES_STORAGE = "whitenoise.storage.CompressedManifestStaticFilesStorage"
-PROJECT_TYPE = getattr(plugin_constants.ProjectType, SERVICE_VARIANT.upper())
-
-
-####################### Plugin Settings ##########################
-
-# Adding plugins for AWS chokes if these are not defined
-ENV_TOKENS = {}
-AUTH_TOKENS = {}
-
-# This is at the bottom because it is going to load more settings after base settings are loaded
-
-# Load aws.py in plugins for reverse compatibility.  This can be removed after aws.py
-# is officially removed.
-plugin_settings.add_plugins(__name__, PROJECT_TYPE, plugin_constants.SettingsType.AWS)
-
-# We continue to load production.py over aws.py
-
-plugin_settings.add_plugins(
-    __name__, PROJECT_TYPE, plugin_constants.SettingsType.PRODUCTION
-)
-
-
-CELERY_IMPORTS = locals().get("CELERY_IMPORTS", [])
-####################### Celery fix ##########################
-# XXX for some reason celery is not registering the bookmarks app
-CELERY_IMPORTS = list(maybe_list(CELERY_IMPORTS)) + [
-    "openedx.core.djangoapps.bookmarks.tasks"
+_settings_modules = [
+    "django",
+    "mysql",
+    "mongo",
+    "caches",
+    "logging",
+    "staticfiles",
+    "media",
+    "storages",
+    "celery",
+    "email",
+    "placeholders",
+    "features",
+    "openedx_platform",
+    "container_env",
+    "plugins",
 ]
 
-########################## This container should never be exposed directly  #######################
-SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
-
-########################## Derive Any Derived Settings  #######################
-
-
-########################## Features #########################
-FEATURES = locals().get("FEATURES", {})
-FEATURES.update(
-    {"ENABLE_COMBINED_LOGIN_REGISTRATION": True, "ENABLE_DISCUSSION_SERVICE": False}
-)
-
-from .container_env import *  # isort:skip
+for setting_module in _settings_modules:
+    setting_module_path = str(Path(__file__).parent / setting_module + ".py")
```

Comment:
We should probably explain why we're using execfile.

Reply:
I added an explanatory comment:

```
# We are using execfile in order to share the current scope so that we
# don't have to redefine and reimport everything in every single settings
# module
```

Is this comprensible/explicit enough ?